frontside/lib/views/home/home.dart

```diff
@@ -1,6 +1,69 @@
+import 'dart:async';
+
 import 'package:flutter/material.dart';
 import 'package:frontside/parts/navbar.dart';
 
+import 'package:frontside/services/places/autocomplete_response.pb.dart';
+import 'package:frontside/services/places/places.dart';
+import 'package:frontside/utils/debouncer.dart';
+
+class PlacesSearchDelegate extends SearchDelegate<AutocompleteEntry?> {
+  final _debouncer = new Debouncer();
+
+  @override
+  Widget buildLeading(BuildContext context) {
+    return IconButton(
+      icon: Icon(Icons.arrow_back),
+      onPressed: () {
+        close(context, null);
+      },
+    );
+  }
+
+  Future<AutocompleteResponse> _getResults() async {
+    final Completer<AutocompleteResponse> _completer = new Completer();
+
+    _debouncer.debounce(() async {
+      _completer.complete(await getPlaces(query));
+    });
+
+    return _completer.future;
+  }
+
+  @override
+  Widget buildResults(BuildContext context) {
+    return FutureBuilder(
+        builder: (BuildContext context,
+            AsyncSnapshot<AutocompleteResponse> response) {
```

Comment:
good practice is to indicate to user to show loader white data is loading or show error on fail and after success show listView

Reply:
I completely agree but I'm going to wait for a design first.